test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -85,51 +131,301 @@ class SslCredentialsTest : public ::testing::Test {
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), common_name);
+  }
+
+  void ExpectPeerSubject(std::shared_ptr<const AuthContext> auth_context,
+                         absl::string_view subject) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_SUBJECT_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), subject);
+  }
+
+  void ExpectPeerSubjectAltName(std::shared_ptr<const AuthContext> auth_context,
+                                const absl::flat_hash_set<std::string>& sans) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_SAN_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), sans.size());
+    absl::flat_hash_set<std::string> observed_sans;
+    for (size_t i = 0; i < properties.size(); ++i) {
+      observed_sans.insert(ToString(properties[i]));
+    }
+    EXPECT_EQ(sans, observed_sans);
+  }
+
+  void ExpectPeerCertPem(std::shared_ptr<const AuthContext> auth_context,
+                         absl::string_view pem) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_PEM_CERT_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), pem);
+  }
+
+  // TODO(mattstevenson88): Check this property when there are >= 2 untrusted
+  // certs in the chain.
+  void ExpectPeerCertChainPem(std::shared_ptr<const AuthContext> auth_context,
+                              absl::string_view pem) {
+    std::vector<grpc::string_ref> properties = auth_context->FindPropertyValues(
+        GRPC_X509_PEM_CERT_CHAIN_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), pem);
+  }
+
+  void ExpectPeerDnsSans(std::shared_ptr<const AuthContext> auth_context,
+                         const absl::flat_hash_set<std::string>& dns_sans) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_PEER_DNS_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), dns_sans.size());
+    absl::flat_hash_set<std::string> observed_sans;
+    for (size_t i = 0; i < properties.size(); ++i) {
+      observed_sans.insert(ToString(properties[i]));
+    }
+    EXPECT_EQ(dns_sans, observed_sans);
+  }
+
+  // TODO(matthewstevenson88): Check this property when there are URI SANs in
+  // the peer leaf cert.
+  void ExpectPeerUriSans(std::shared_ptr<const AuthContext> auth_context,
+                         const absl::flat_hash_set<std::string>& uri_sans) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_PEER_URI_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), uri_sans.size());
+    absl::flat_hash_set<std::string> observed_sans;
+    for (size_t i = 0; i < properties.size(); ++i) {
+      observed_sans.insert(ToString(properties[i]));
+    }
+    EXPECT_EQ(uri_sans, observed_sans);
```

Comment:
nit: reverse the order to be consistent in this test file (looks like all the `sans` related ones are reversed from the others)

Reply:
Done.